task/06-conditions-n-loops-tasks.js

```diff
@@ -332,7 +377,27 @@ function isBracketsBalanced(str) {
  *
  */
 function timespanToHumanString(startDate, endDate) {
-    throw new Error('Not implemented');
+    const time = (endDate - startDate) / 1000;
+    const minutes = 60;
+    const hours = minutes * 60;
+    const days = hours * 24;
+    const months = days * 30;
+    const years = months * 12;
+
+    function round(num) {
+        return num - Math.floor(num) > 0.5 ? Math.round(num) : Math.floor(num);
+    }
+
+    return time > 545 * days ? `${round(time / years)} years ago` :
+        time > 345 * days ? 'a year ago' :
+            time > 45 * days ? `${round(time / months)} months ago` :
+                time > 25 * days ? 'a month ago' :
+                    time > 36 * hours ? `${round(time / days)} days ago` :
+                        time > 22 * hours ? 'a day ago' :
+                            time > 90 * minutes ? `${round(time / hours)} hours ago` :
+                                time > 45 * minutes ? 'an hour ago' :
+                                    time > 90 ? `${round(time / 60)} minutes ago` :
+                                        time > 45 ? 'a minute ago' : 'a few seconds ago';
```

Comment:
avoid this code style, if-else is more readable than multiple conditional operators ? :